src/segmented-control/segment.scss

```diff
@@ -38,65 +38,99 @@ $selected-color: awsui.$color-text-segment-active;
   @include styles.text-flex-wrapping;
   @include styles.font-button;
   letter-spacing: 0.25px;
-  border-block: styles.$control-border-width solid $default-border-color;
-  border-inline: styles.$control-border-width solid $default-border-color;
   padding-block: styles.$control-padding-vertical;
   padding-inline: awsui.$space-button-horizontal;
-  border-inline-end-width: 0;
   background: $default-background;
   color: $default-color;
   overflow: visible;
+
   &:focus {
     outline: none;
   }
-  @for $i from 1 through 6 {
-    &:nth-child(#{$i}) {
-      grid-column: $i;
-      -ms-grid-column: $i;
-    }
+
+  &.disabled {
+    background: $disabled-background;
+    border-color: $disabled-border-color;
+    color: $disabled-color;
   }
+
   @include focus-visible.when-visible {
     @include styles.focus-highlight(awsui.$space-segmented-control-focus-outline-gutter);
   }
-  &:last-child {
-    border-inline-end-width: styles.$control-border-width;
-    border-start-start-radius: 0;
-    border-start-end-radius: awsui.$border-radius-button;
-    border-end-start-radius: 0;
-    border-end-end-radius: awsui.$border-radius-button;
-  }
-  &:first-child {
+
+  &.refresh {
+    border-inline: none;
+    border-block: none;
     border-start-start-radius: awsui.$border-radius-button;
-    border-start-end-radius: 0;
+    border-start-end-radius: awsui.$border-radius-button;
     border-end-start-radius: awsui.$border-radius-button;
-    border-end-end-radius: 0;
+    border-end-end-radius: awsui.$border-radius-button;
+    padding-block: awsui.$space-scaled-xxs;
+    padding-inline: awsui.$space-scaled-m;
+    position: relative;
+    block-size: calc(100% - 4px);
```

Comment:
there are a lot of "magic numbers" here - should some of them be calculated based on spacing tokens and/or eachother?

Reply:
Thanks @gethinwebster,
I updated them. 
- For paddings, it's now based off of the paddings used in button component
- `$segment-divider-gap: awsui.$space-static-xxs;` is now base size without using hard code value